common/lib/xmodule/xmodule/fields.py

```diff
@@ -252,3 +253,19 @@ def enforce_type(self, value):
             return value
 
         return self.from_json(value)
+
+
+class ScoreField(JSONField):
+    def from_json(self, value):
+        if value is None:
+            return value
+        if isinstance(value, Score):
+            return value
+
+        keys = value.keys()
+        if 'raw_earned' not in keys or 'raw_possible' not in keys:
+            raise TypeError('Scores must contain only a raw earned and raw possible value')
+
+        return Score(raw_earned=value['raw_earned'], raw_possible=value['raw_possible'])
```

Comment:
I think this could justifiably be added to the XBlock repo, not just this file here, but this is the only place it is used so I didn't open the separate PR.